rooms.js

```diff
@@ -163,6 +165,95 @@ var Room = (function () {
 		}
 		return true;
 	};
+	//mute handling
+	Room.prototype.runMuteTimer = function () {
+		if (this.muteTimer) return;
+		if (this.mutedUsers[0].time < Date.now() + 5) {
+			// this prevents lag from bugging the timer
+			this.mutedUsers[0].time = Date.now() + 5;
+		}
+		var self = this;
+		this.muteTimer = setTimeout(function () {
+			var user = Users(self.mutedUsers[0].userid);
```

Comment:
This seems brittle, to me. You should just have a function that checks to see if the mute closest to expiring has expired yet, and then wherever necessary set the mute timer to run that.